File: frontend/src/types/product.ts
export interface Product {
  id: string;
  sku: string;
  name: string;
  slug: string;
  tagline?: string;
  description?: string;
  base_price: number;
  selling_price: number;
  gst_percent?: number;
  specs?: Record<string, unknown>;
  warranty_years?: number;
  weight_kg?: number;
  is_assembly_required?: boolean;
  is_hero: boolean;
  is_b2b_available: boolean;
  ar_model_url?: string;
  meta_title?: string;
  meta_description?: string;
  category?: Category;
  variants?: ProductVariant[];
  images?: ProductImage[];
}

export interface ProductVariant {
  id: string;
  sku_suffix: string;
  color: string;
  material?: string;
  price_adjustment: number;
}

export interface ProductImage {
  id: string;
  url: string;
  alt_text?: string;
  sort_order: number;
  is_primary: boolean;
}

export interface Category {
  id: string;
  name: string;
  slug: string;
  description?: string;
}

export interface CartItem {
  product: Product;
  variant?: ProductVariant;
  quantity: number;
}

File: backend/app/schemas/product.py
"""Pydantic schemas for product endpoints."""

from decimal import Decimal
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class CategoryOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    name: str
    slug: str
    description: Optional[str] = None


class ProductImageOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    url: str
    alt_text: Optional[str] = None
    sort_order: int
    is_primary: bool


class ProductVariantOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    sku_suffix: str
    color: str
    material: Optional[str] = None
    price_adjustment: Decimal


class ProductListOut(BaseModel):
    """Lightweight product for listing pages."""
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    sku: str
    name: str
    slug: str
    tagline: Optional[str] = None
    base_price: Decimal
    selling_price: Decimal
    is_hero: bool
    is_b2b_available: bool
    ar_model_url: Optional[str] = None


class ProductDetailOut(ProductListOut):
    """Full product detail for PDP."""
    description: Optional[str] = None
    specs: Optional[dict] = None
    warranty_years: int
    weight_kg: Optional[Decimal] = None
    is_assembly_required: bool
    gst_percent: Decimal
    meta_title: Optional[str] = None
    meta_description: Optional[str] = None
    category: Optional[CategoryOut] = None
    variants: list[ProductVariantOut] = []
    images: list[ProductImageOut] = []


class ProductCreate(BaseModel):
    sku: str
    name: str
    slug: str
    tagline: Optional[str] = None
    description: Optional[str] = None
    base_price: Decimal
    selling_price: Decimal
    cost_price: Decimal
    gst_percent: Decimal = Decimal("18.00")
    specs: Optional[dict] = None
    warranty_years: int = 1
    weight_kg: Optional[Decimal] = None
    is_assembly_required: bool = True
    is_hero: bool = False
    is_b2b_available: bool = True
    ar_model_url: Optional[str] = None
    category_id: Optional[UUID] = None


class ProductUpdate(BaseModel):
    name: Optional[str] = None
    tagline: Optional[str] = None
    description: Optional[str] = None
    base_price: Optional[Decimal] = None
    selling_price: Optional[Decimal] = None
    cost_price: Optional[Decimal] = None
    specs: Optional[dict] = None
    is_hero: Optional[bool] = None
    is_active: Optional[bool] = None
    ar_model_url: Optional[str] = None
